### PR TITLE
Docs: Update annotations API docs to include required field

### PR DESCRIPTION
**Pull request by HarryTennent**
**What this PR does / why we need it**: A minor change to include the required fields when creating a new annotation via the annotations API. The error message when fields are missing is not descriptive and there is no indication in the current documentation (that I can see anyway) that would indicate these fields are required.

**Which issue(s) this PR fixes**: No issue found for this. If this is a requirement for raising a PR please let me know and I'll create one. 

Fixes #

**Special notes for your reviewer**: This is my first PR to the Grafana repo, I've read through `contributing.md` and the doc contribution readme, but please let me know if I've missed something.

### CODE REVIEW

**Review by brendamuir — APPROVED** [review]

**Comment by chri2547** [comment]
Would someone from @grafana/backend-platform please review and approve this PR.

File: docs/sources/developers/http_api/annotations.md [hunk 1]
```diff
@@ -121,6 +121,12 @@ See note in the [introduction]({{< ref "#annotations-api" >}}) for an explanatio
 | ------------------ | ----------------------- |
 | annotations:create | annotations:type:<type> |
 
+
+**Required JSON Body Fields**
+  - `time`: epoch datetime in milliseconds.
+  - `timeEnd`: epoch datetime in milliseconds - must be greater than `time`.
```
Comment (papagian) [inline docs/sources/developers/http_api/annotations.md:127]: I don't think that those two fields are required
I can successfully create an annotation without them:
```
curl -X 'POST'   'http://<user>:<password>@localhost:3000/api/annotations'   -H 'accept: application/json'   -H 'Content-Type: application/json'   -d '{
  "dashboardId": 0,
  "dashboardUID": "string",
  "data": {},
  "panelId": 0,
  "tags": [
    "string"
  ],
  "text": "string"
}'
{"id":37712,"message":"Annotation added"}
```

**Review by papagian — COMMENTED** [review]
We have to update also the swagger specification; therefore we have to add the necessary swagger annotation in [PostAnnotationsCmd](https://github.com/grafana/grafana/blob/0c8974375956535e15741bc67dd4d34038a8557d/pkg/api/dtos/annotations.go#L5)
```
// swagger:model
type PostAnnotationsCmd struct {
	DashboardId  int64            `json:"dashboardId"`
	DashboardUID string           `json:"dashboardUID,omitempty"`
	PanelId      int64            `json:"panelId"`
	Time         int64            `json:"time"`
	TimeEnd      int64            `json:"timeEnd,omitempty"` // Optional
	// required: true
	Text         string           `json:"text"`
	Tags         []string         `json:"tags"`
	Data         *simplejson.Json `json:"data"`
}
```

If we do so and we re-generate the specification [the model](https://github.com/grafana/grafana/blob/0c8974375956535e15741bc67dd4d34038a8557d/public/api-spec.json#L11881) becomes:
```
    "PostAnnotationsCmd": {
      "type": "object",
      "required": [
        "text"
      ],
      "properties": {
        "dashboardId": {
          "type": "integer",
          "format": "int64"
        },
        "dashboardUID": {
          "type": "string"
        },
        "data": {
          "$ref": "#/definitions/Json"
        },
        "panelId": {
          "type": "integer",
          "format": "int64"
        },
        "tags": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "text": {
          "type": "string"
        },
        "time": {
          "type": "integer",
          "format": "int64"
        },
        "timeEnd": {
          "type": "integer",
          "format": "int64"
        }
      }
    },
```

File: docs/sources/developers/http_api/annotations.md (hunk 1 above)
Comment (HarryTennent) [reply docs/sources/developers/http_api/annotations.md:127]: Ah, I was getting an unrelated error when trying to add without those fields - I'll remove them from the change request. Thanks for the sanity check

**Review by chri2547 — COMMENTED** [review]
The docs look fine to me. Please go ahead and merge upon dev review and approval.

**Pull request merged**